rmf_fleet_adapter/src/rmf_fleet_adapter/agv/RobotContext.cpp

```diff
@@ -393,6 +393,20 @@ uint32_t RobotContext::current_mode() const
   return _current_mode;
 }
 
+//==============================================================================
+void RobotContext::action_executor(
+  RobotUpdateHandle::ActionExecutor action_executor)
+{
+  if (action_executor != nullptr)
```

Comment:
I'm not actually sure that we should block this. What if the user is trying to erase its action executor for some reason?

I recommend instead of blocking, we simply emit a warning that the action executor is being nullified which will lead to an automatic critical task error if a task requires an action to be performed while this is still set to null.